File: benchmarks/values/string_count.go
package main

import (
	"fmt"
	"strings"
	"time"
)

func main() {
	t0 := time.Now()

	base := "the quick brown fox jumps over the lazy dog "
	text := strings.Repeat(base, 2500000)

	count := 0
	pos := 0
	for {
		idx := strings.Index(text[pos:], "fox")
		if idx == -1 {
			break
		}
		count++
		pos += idx + 3
	}

	elapsed := time.Since(t0)
	fmt.Println(count)
	fmt.Printf("elapsed: %.3fs\n", elapsed.Seconds())
}

File: benchmarks/values/string_count.rs
use std::time::Instant;

fn main() {
    let t0 = Instant::now();

    let base = "the quick brown fox jumps over the lazy dog ";
    let text = base.repeat(2500000);

    let mut count = 0;
    let mut pos = 0;
    let needle = "fox";
    let needle_len = needle.len();
    while pos <= text.len() - needle_len {
        match text[pos..].find(needle) {
            Some(idx) => {
                count += 1;
                pos += idx + needle_len;
            }
            None => break,
        }
    }

    let elapsed = t0.elapsed();
    println!("{}", count);
    println!("elapsed: {:.3}s", elapsed.as_secs_f64());
}
